eng/validation/update-channel.ps1

```diff
@@ -15,34 +15,45 @@ function Get-Headers([string]$accept, [string]$barToken) {
 
 $arcadeSdkPackageName = 'Microsoft.DotNet.Arcade.Sdk'
 $arcadeSdkVersion = $GlobalJson.'msbuild-sdks'.$arcadeSdkPackageName
-$getAssetApiEndpoint = "$maestroEndpoint/api/assets?name=$arcadeSdkPackageName&version=$arcadeSdkVersion&api-version=2018-07-16"
+$getAssetsApiEndpoint = "$maestroEndpoint/api/assets?name=$arcadeSdkPackageName&version=$arcadeSdkVersion&api-version=2018-07-16"
 $headers = Get-Headers 'text/plain' $barToken
 
 # Get the Microsoft.DotNet.Arcade.Sdk with the version $arcadeSdkVersion so we can get the id of the build
-$assets = Invoke-WebRequest -Uri $getAssetApiEndpoint -Headers $headers | ConvertFrom-Json
-
-if ($assets) {
-    # Today, there shouldn't be more that one build linked to a given Arcade SDK version, but if in the future we only bump the version if
-    # changes were done that could be a possibility. Also, since we already validated version $arcadeSdkVersion we move all the builds that
-    # produced it to channel $channelId
-    foreach ($asset in $assets) {
-        try {
-            $buildId = $asset[0].'buildId'
-
-            $postBuildIntoChannelApiEndpoint = "$maestroEndpoint/api/channels/$channelId/builds/$buildId/?name=$arcadeSdkPackageName&version=$arcadeSdkVersion&api-version=2018-07-16"
-            $headers = Get-Headers 'application/json' $barToken
-        
-            Write-Host "POSTing to $postBuildIntoChannelApiEndpoint..."
-            $postResponse = Invoke-WebRequest -Uri $postBuildIntoChannelApiEndpoint -Headers $headers -Method Post
-            Write-Host "Build '$buildId' was successfully added to channel '$channelId'"
-         }
-        catch {
-            Write-Host $_
-            Write-Host $_.ScriptStackTrace
-            exit 1
-        }
-    }
-} else {
+$assets = Invoke-WebRequest -Uri $getAssetsApiEndpoint -Headers $headers | ConvertFrom-Json
+
+if (!$assets) {
     Write-Host "Asset '$arcadeSdkPackageName' with version $arcadeSdkVersion was not found"
     exit 1
-}
\ No newline at end of file
+}
+
+if ($assets.Count > 1) {
+    Write-Host "More than 1 asset matched the version '$arcadeSdkVersion' of Microsoft.DotNet.Arcade.Sdk. This is not normal. Stopping execution..."
+    exit 1
+}
+
+$getChannelsEndpoint = "$maestroEndpoint/api/channels?api-version=2018-07-16"
+$channels = Invoke-WebRequest -Uri $getChannelsEndpoint -Headers $headers | ConvertFrom-Json
+$matchedChannel = $($channels | Where-Object -Property "name" -Value "${targetChannelName}" -EQ | Select-Object -Property id)
+
+if (!$matchedChannel) {
+    Write-Host "Channel with name '$targetChannelName' was not found..."
+    exit 1    
+}
+
+$channelId = $matchedChannel.id
+
+try {
```

Comment:
Why the try/catch here but not around the other calls?

Reply:
When a build already exists in a channel we are returning 500 which for Invoke-WebRequest is considered an exception. Given the refactor we could just let the call fail since we are not in a loop anymore

Reply:
Okay.  I have an issue I just opened re: that issue, will fix.